lib/mayaUsd/fileio/primUpdater.cpp

```diff
@@ -73,15 +77,32 @@ bool UsdMayaPrimUpdater::editAsMaya(const UsdMayaPrimUpdaterContext& context) {
 bool UsdMayaPrimUpdater::discardEdits(const UsdMayaPrimUpdaterContext& context)
 {
     MObject objectToDelete = getMayaObject();
-    if (!objectToDelete.isNull()) {
-        MGlobal::deleteNode(objectToDelete);
+    if (objectToDelete.isNull())
+        return true;
+
+    auto& undoInfo = UsdUndoManager::instance().getUndoInfo();
+
+    MFnDependencyNode depNode(objectToDelete);
+
+    MStatus status = NodeDeletionUndoItem::deleteNode(
+        "Discard edits delete individual pulled node",
+        depNode.absoluteName(),
+        objectToDelete,
+        undoInfo);
+
+    if (status != MS::kSuccess) {
+        TF_WARN("Discard edits: cannot delete node.");
+        return false;
     }
-    return true;
+
+    return status == MS::kSuccess;
 }
 
 bool UsdMayaPrimUpdater::pushEnd(const UsdMayaPrimUpdaterContext& context)
 {
-    return discardEdits(context);
+    // Nothing. We will rely on the Maya scene cleaup code
```

Comment:
What is the Maya scene cleanup code?

Reply:
Oh it's the code that goes through the Maya nodes in PrimUpdaterManager. I'll clarify the comment to say that instead.